src/redisearch_rs/encode_decode/src/varint/vector_writer.rs

```diff
@@ -32,7 +32,6 @@ impl VectorWriter {
     ///
     /// The number of bytes written to the vector.
     pub fn write(&mut self, value: u32) -> std::io::Result<usize> {
-        self.buffer.reserve_exact(16);
```

Comment:
So now the call to `write` will reallocate the buffer if needed, right?

Reply:
yes, since it's being passed as the `&mut Vec<u8>` instead of as a mutable slice (`&mut [u8]`).